Sustainsys.Saml2/WebSSO/AcsCommand.cs

```diff
@@ -88,6 +95,17 @@ public CommandResult Run(HttpRequestData request, IOptions options)
             throw new NoSamlResponseFoundException();
         }
 
+        private static IdentityProvider GetIdpContext(XmlElement xml, IOptions options)
+        {
+            var entityId = new EntityId(xml["Issuer", Saml2Namespaces.Saml2Name].GetTrimmedTextIfNotNull());
+
+            IdentityProvider identityProvider;
+
+            options.IdentityProviders.TryGetValue(entityId, out identityProvider);            
```

Comment:
Use the notification instead, to allow consumers to overload Idp lookup (which allows to bypass the IdentityProviders collection completely).

Reply:
I have updated the code to use Notification instead.